components/motor/gpio/setup.go

```diff
@@ -213,7 +220,7 @@ func createNewMotor(
 				return nil, err
 			}
 		default:
-			m, err = setupMotorWithControls(ctx, basic, e, cfg, logger)
+			m, err = setupMotorWithControls(ctx, m, e, cfg, logger)
```

Comment:
all the changes in controlled motor come from this change. I wanted to remove the basic casting and try to keep everything at the motor interface level